packages/plugin-phone/src/call.js

```diff
@@ -558,13 +578,13 @@ const Call = SparkPlugin.extend({
 
     this.media.end();
 
-    if (!this.locus) {
-      if (this.locusJoinInFlight) {
-        this.logger.info(`call: no locus, waiting for rest call to complete before hanging up`);
-        return this.when(`change:locus`)
-          .then(() => this.hangup());
-      }
+    if (this.locusJoinInFlight) {
+      this.logger.info(`call: locus join in flight, waiting for rest call to complete before hanging up`);
+      return this.when(`change:locusJoinInFlight`)
+        .then(() => this.hangup());
```

Comment:
line 581 states that join in flight is true, so if a change event fires, it can only become false.